File: 1323-maximum-69-number/1323-maximum-69-number.cs
public class Solution {
    public int Maximum69Number (int num) {
        var mask = GetMask(num);    // for number like 9669, mask will be 1000
        var ans = 0;
        var flip = false;   // has flip been done?

        while(num > 0){
            var digit = num / mask;
            if(!flip && digit == 6){
                ans += mask * 9;
                flip = true;
            }
            else
                ans += mask * digit;
            
            num = (num % (digit * mask));
            mask /= 10;
        }

        return ans;
    }

    private int GetMask(int num){
        var mask = 1;
        while(num >= 10){
            num /= 10;
            mask *= 10;
        }
        return mask;
    }
}

File: 1323-maximum-69-number/1323-maximum-69-number.py
class Solution:
    def maximum69Number (self, num: int) -> int:
        mask = self.getMask(num)    # for number like 9669, mask will be 1000
        ans = 0
        flip = False   # has flip been done?

        while num > 0:
            digit = num // mask
            if not flip and digit == 6:
                ans += mask * 9
                flip = True
            else:
                ans += mask * digit
            
            num = (num % (digit * mask))
            mask //= 10

        return ans

    def getMask(self, num):
        mask = 1
        while num >= 10:
            num //= 10
            mask *= 10
        return mask